pkg/activator/handler/handler.go

```diff
@@ -115,3 +118,14 @@ func (s *statusCapture) WriteHeader(statusCode int) {
 	s.statusCode = statusCode
 	s.ResponseWriter.WriteHeader(statusCode)
 }
+
+// Hijack calls Hijack() on the wrapped http.ResponseWriter if it implements
+// http.Hijacker interface, which is required for net/http/httputil/reverseproxy
+// to handle connection upgrade/switching protocol.  Otherwise returns an error.
+func (s *statusCapture) Hijack() (net.Conn, *bufio.ReadWriter, error) {
+	hj, ok := s.ResponseWriter.(http.Hijacker)
+	if !ok {
+		return nil, nil, fmt.Errorf("Wrapped writer type %T can't be hijacked", s.ResponseWriter)
+	}
+	return hj.Hijack()
```

Comment:
Can this be pulled into a helper method so that we can do:

```golang
func (s *statusCapture) Hijack() (net.Conn, *bufio.ReadWriter, error) {
	return HijackIfPossible(s.ResponseWriter)
}
```

Reply:
Even though unrelated to what's currently there, I wonder if `knative/pkg/websocket` might be the best place for that.  Follow up?